### PR TITLE
feat(headless): add #id option to tab controller; send originLevel2 as #tab parameter

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/KIT-407

### CODE REVIEW

File: packages/headless/src/controllers/tab/headless-tab.ts [hunk 1]
```diff
@@ -27,6 +28,11 @@ export interface TabOptions {
    * `@objecttype==Message`
    */
   expression: string;
+
+  /**
+   * A unique identifier for the tab. The value will used as the originLevel2 when the tab is active.
+   */
+  id?: string;
```
Comment (samisayegh) [inline packages/headless/src/controllers/tab/headless-tab.ts:35]: What do you think of `id` as the option name compared to `name` or other?

Comment (nbuissoncoveo) [reply packages/headless/src/controllers/tab/headless-tab.ts:35]: I think `id` is more fitting for tabs rather than `name`

Comment (btaillon) [reply packages/headless/src/controllers/tab/headless-tab.ts:35]: I also think `id` is more fitting, but I think we should give an example in the description of what that could look like.

Usually, when a framework asks me for an `id` and doesn't provide examples, I have to go looking for whether there's validation on it (e.g., no spaces? no dashes? no special characters?).

Perhaps we should also include a link to some doc explaining what originLevel2 is.

**Review by nbuissoncoveo — APPROVED** [review]

**Review by ThibodeauJF — APPROVED** [review]

File: packages/headless/src/controllers/tab/headless-tab.ts [hunk 2]
```diff
@@ -37,8 +43,9 @@ export interface TabInitialState {
   isActive: boolean;
 }
 
-const optionsSchema = new Schema({
-  expression: new StringValue(),
+const optionsSchema = new Schema<Required<TabOptions>>({
+  expression: new StringValue({required: true, emptyAllowed: true}),
+  id: new StringValue({required: false, emptyAllowed: false}),
```
Comment (olamothe) [inline packages/headless/src/controllers/tab/headless-tab.ts:48]: We could set the default to `default` ... 🤔  It matches what we set  for origin level 2 by default.

And we could change the warning message to verify that, if you use a tab, you should set it to something else than 'default' ?

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/controllers/tab/headless-tab.ts (hunk 2 above)
Comment (samisayegh) [reply packages/headless/src/controllers/tab/headless-tab.ts:48]: Sounds like an improvement 👍

Comment (samisayegh) [reply packages/headless/src/controllers/tab/headless-tab.ts:48]: That would be an improvement 👍

**Pull request merged**